PopApp/Store/configureStore.js

```diff
@@ -1,10 +1,44 @@
 import { createStore } from 'redux';
+import { persistStore, persistCombineReducers } from 'redux-persist';
+import storage from 'redux-persist/lib/storage';
```

Comment:
I had not think about this problem. I cannot find information about if redux-persist with asyncStorage is protected againt reordering.

Reply:
The problem is not so much about reordering, but rather: "what information will the UI display?"

Image I sent a `requestCreateLao` and my UI connected to it. Then I `requestCreateLao` another LAO. Here's what will happen when the server answers my second request (the function `handleAnswer` will be called with the `answer` as parameter) :

```javascript
const handleAnswer = (answer) => {

    if (answer.isOfTypeCreateLao()) {
        if (answer.isPositive()) {
            // store newly created LAO (second one) into the local storage
            // this line would be `async` if we use the `AsyncStorage`
            getStore().dispatch({ type: 'SET_CURRENT_LAO', value: newLao });
            // tell the UI to display the new LAO
            OrganizerUI.launch(); /* ??? */

        } else {
            // do stuff, e.g. send the request again
        }
    } else { /* ... */ }
}
```

ℹ️  this is pseudocode. If you want to see how it really works, check the function `handleServerAnswer` in `websocket/WebsocketAnswer.js` (especially lines 63 to 74)


The problem lies where `/* ??? */` is situated. Which LAO will the organizer ui load (since Organizer ui probably uses `getStore().getState()`)? The old one (if the store didn't happen yet, or the new one)?

Note that I agree that in this particular case, we could pass the newly created LAO as a parameter for the launch method... But this it is not (in my opinion) a clean way to do it... Moreover, this problem could occur somewhere else in the organizer UI and passing the newly created LAO to each of them would not be a good idea at all.